netmiko/cisco/cisco_xr.py

```diff
@@ -105,36 +114,74 @@ def commit(
         else:
             command_string = "commit"
 
+        if force:
+            command_string = command_string.replace("commit", "commit force")
+        if best_effort:
+            command_string = command_string.replace("commit", "commit best-effort")
+        if replace:
+            command_string = command_string.replace("commit", "commit replace")
+        print(f"command string is {command_string}")
+
         # Enter config mode (if necessary)
         output = self.config_mode()
 
-        # IOS-XR might do this:
-        # This could be a few minutes if your config is large. Confirm? [y/n][confirm]
-        new_data = self._send_command_str(
-            command_string,
-            expect_string=r"(#|onfirm)",
-            strip_prompt=False,
-            strip_command=False,
-            read_timeout=read_timeout,
-        )
-        if "onfirm" in new_data:
-            output += new_data
+        if replace:
+            new_data = self._send_command_str(
+                command_string,
+                expect_string=r"This commit will replace or remove the entire running configuration",
+                strip_prompt=False,
+                strip_command=False,
+                read_timeout=read_timeout,
+            )
+            if "This commit will replace or remove the entire running configuration" in new_data:
+                output += new_data
+                new_data = self._send_command_str(
+                    "yes",
+                    expect_string=r"#",
+                    strip_prompt=False,
+                    strip_command=False,
+                    read_timeout=read_timeout,
+                )
+        else:
+            # IOS-XR might do this:
+            # This could be a few minutes if your config is large. Confirm? [y/n][confirm]
             new_data = self._send_command_str(
-                "y",
-                expect_string=r"#",
+                command_string,
+                expect_string=r"(#|onfirm)",
                 strip_prompt=False,
                 strip_command=False,
                 read_timeout=read_timeout,
             )
+            if "onfirm" in new_data:
+                output += new_data
+                new_data = self._send_command_str(
+                    "y",
+                    expect_string=r"#",
+                    strip_prompt=False,
+                    strip_command=False,
+                    read_timeout=read_timeout,
+                )
         output += new_data
         if error_marker in output:
             raise ValueError(f"Commit failed with the following errors:\n\n{output}")
         if alt_error_marker in output:
+            if commit_error_dialog_dict is not None and alt_error_marker in commit_error_dialog_dict:
+                marker_value = commit_error_dialog_dict[alt_error_marker]
+                output += new_data
+                new_data += self._send_command_str(
```

Comment:
this should be new_data = and not += . better to just use output everywhere and remove new_data completely